docs/compilers/CSharp/Compiler Breaking Changes - DotNet 7.md

```diff
@@ -1,5 +1,32 @@
 # This document lists known breaking changes in Roslyn after .NET 6 all the way to .NET 7.
 
+## Caller of an async local function cannot expect any of the callee�s code to execute.
+
+***Introduced in Visual Studio 2022 version 17.5***
+
+See https://github.com/dotnet/roslyn/issues/43697 for the rational.
```

Comment:
Consider adding a one sentence summary, maybe something like:
For definite assignment analysis, invocations of async local functions are no longer treated as being awaited and therefore the local function is not considered executed.

Let's send an email as FYI to compat council for new compat break.

It would be good to add a note to the [spec](https://github.com/dotnet/csharplang/blob/main/proposals/csharp-7.0/local-functions.md) even though it's an older/incomplete document, and tag Rex on the edit (so that definite assignment rules in standard can reflect this change as appropriate).